drivers/periph_common/i2c.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2018 Mesotic SAS <dylan.laduranty@mesotic.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers
+ * @{
+ *
+ * @file
+ * @brief       common I2C function fallback implementations
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "cpu.h"
+#include "periph/i2c.h"
+
+#ifdef I2C_NUMOF
+
+#ifdef PERIPH_I2C_NEED_READ_REG
+int i2c_read_reg(ic2_t dev, uint16_t addr, uint16_t reg,
+                 void *data, uint8_t flags)
+{
+	return i2c_read_regs(dev, addr, reg, data, 1, flags);
+}
+#endif /* PERIPH_I2C_NEED_READ_REG */
+
+#ifdef PERIPH_I2C_NEED_READ_REGS
+int i2c_read_regs(ic2_t dev, uint16_t addr, uint16_t reg,
+                  void *data, size_t len, uint8_t flags);
+{
+	uint16_t data = reg;
```

Comment:
Why do you need `data` and not just use `reg` in other places ?
By the way, this is confusing since `data` is also used as `void *` for the output buffer